### PR TITLE
feat(app): Enable buildroot updates by default

### LINKED ISSUE #3822: Balena removal: Enable buildroot updates by default
## overview

This ticket is for tracking the removal of the app's old wheel based update system in favor of the buildroot update system

~Blocked by #3820~

## acceptance criteria

- [ ] Ensure bootstrap wheels are up to date
- [ ] Update and remove UI components
    - [ ] Remove `app/src/components/RobotSettings/UpdateRobot/`
    - [ ] Update `app/src/components/RobotSettings/InformationCard.js`
    - [ ] Update `app/src/pages/Robots/RobotSettings.js`
    - [ ] Update `app/src/components/ConnectPanel/RobotItem.js`
- [ ] Remove `app/src/shell/api-update.js`
- [ ] Remove `app/src/http-api-client/server.js`
- [ ] Remove `app-shell/src/api-update.js`
    - [ ] Update `app-shell/src/preload.js` accordingly
- [ ] Remove `devInternal.buildRootEnabled` from `Config` typedef

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR closes #3822 by removing old update files/logic and enabling buildroot update by default. 

## changelog
- refactor: Remove old UpdateRobot
- refactor: Remove FF and enable BuildrootUpdate by default
- refactor: Update notification dots in robot list and nav button
- refactor: Remove api-update files from app and app-shell
- refactor: Update .whl files for migration
- refactor: Remove all actions/selectors/state from server.js except restart

## review requests
```
make -C app dev
```
**Migrating Balena Robot**

- [ ] Correct button text in RobotInfoCard for upgrade/downgrade/reinstall
- [ ] Notification dot for robots that robot version != app version
- [ ] Notification dot shows in navbar button when connected robot version != app version
- [ ] Auto pop up for update when first viewing a robot that robot version != app version
- [ ] Migration works (step 1 of 2)
- [ ] Update works (step 2 of 2)
- [ ] Upgrade downgrade/reinstall/release notes screens behave as expected

**Buildroot Robots**

- [ ] Correct button text in RobotInfoCard for upgrade/downgrade/reinstall
- [ ] Notification dot for robots that robot version != app version
- [ ] Notification dot shows in navbar button when connected robot version != app version
- [ ] Auto pop up for update when first viewing a robot that robot version != app version
- [ ] Migration works (step 1 of 2)
- [ ] Update works (step 2 of 2)
- [ ] Upgrade downgrade/reinstall/release notes screens behave as expected

**Optional (But Useful) Tests**
- [ ] Upgrade 
- [ ] Downgrade
- [ ] Reinstall
- [ ] Download in progress modal
- [ ] Update Error modal

### CODE REVIEW

File: app/src/config/index.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ export const DEV_INTERNAL_FLAGS: Array<DevInternalFlag> = [
   'tempdeckControls',
   'enableThermocycler',
   'enablePipettePlus',
-  'enableBuildRoot',
```
Comment (mcous) [inline app/src/config/index.js]: This same change needs to be made in the types file:

https://github.com/Opentrons/opentrons/blob/e025ba66714d88914dd168d79567a39505c1a7cf/app/src/config/types.js#L10-L15

Comment (Kadee80) [reply app/src/config/index.js]: damn i thought i got that one, will remove

File: app-shell/src/preload.js [hunk 2]
```diff
@@ -15,7 +15,4 @@ global.APP_SHELL_REMOTE = {
   CURRENT_RELEASE_NOTES,
   INITIAL_CONFIG: cloneDeep(getConfig()),
   INITIAL_ROBOTS: getRobots(),
-
-  // TODO(mc, 2019-08-05): remove when we remove __buildrootEnabled FF
-  apiUpdate: remote.require('./api-update'),
```
Comment (mcous) [inline app-shell/src/preload.js]: Same change needs to be made in the `remote` mock:

https://github.com/Opentrons/opentrons/blob/d5f3fe31350d6e7e7322a0812ed8f4f22be69d0b/app/src/shell/__mocks__/remote.js#L13-L16

File: app/src/components/nav-bar/NavButton.js [hunk 3]
```diff
@@ -33,7 +36,11 @@ function mapStateToProps(state: State, ownProps: OP): $Exact<Props> {
   const isProtocolDone = robotSelectors.getIsDone(state)
   const isConnected =
     robotSelectors.getConnectionStatus(state) === robotConstants.CONNECTED
-  const robotNotification = getConnectedRobotUpgradeAvailable(state)
+  const connectedRobot = getConnectedRobot(state)
```
Comment (mcous) [inline app/src/components/nav-bar/NavButton.js]: Nitpick, but having both `isConnected` and `connectedRobot` is redundant. We can get rid of `isConnected` and use `connectedRobot != null` instead

**Pull request merged**